src/ScanPatron/ScanPatron.js

```diff
@@ -8,22 +8,45 @@ import ScanForm from './ScanForm';
 const ScanPatron = ({ mutator, resources, stripes }) => {
   const [scannedPatronDetails, setScannedPatronDetails] = useState();
   const [patronRRAPermission, setPatronRRAPermission] = useState();
+  const [readingRoomName, setReadingRoomName] = useState();
+  const [loading, setLoading] = useState(false);
+
+  const currServicePointId = useRef();
+
+  const currentReadingRoomName = useCallback(async () => {
+    if (currServicePointId.current) {
+      const query = `servicePoints.id==${currServicePointId.current}`;
+      const roomDetails = await mutator.readingRoom.GET({ params: { query } });
+      setReadingRoomName(roomDetails[0]?.name);
+    }
+  }, [mutator.readingRoom]);
+
+  useEffect(() => {
+    const currSPIdFromProps = stripes?.user?.user?.curServicePoint?.id;
```

Comment:
Also, with `react-query` you won't need to store `currServicePointId` ref - you can pass service point id to the query keys and it will automatically re-fetch the data when the id changes

Reply:
@BogdanDenis 
Do you agree if I create a tech debt in order to address your comment as we have already made a huge journey already with stripes connect?

Reply:
I have moved to useQuery here. Please re-review.